server/game/triggeredability.js

```diff
@@ -53,23 +53,32 @@ class TriggeredAbility extends BaseAbility {
     }
 
     createContext(event) {
-        return new TriggeredAbilityContext({
+        const context = new TriggeredAbilityContext({
             ability: this,
             event: event,
             game: this.game,
             source: this.card,
             player: this.playerFunc()
         });
+        const listener = this.when[event.name];
+        if (typeof listener === 'function') {
+            context.isTriggered = listener(event, context);
+            return context;
+        } else {
+            const aggregates = this.getAggregatedEvents(event, listener, context);
+            // Concat all aggregates which have passed the condition (there could be multiple)
+            context.events = [...aggregates.map((a) => a.events)];
+            context.isTriggered = context.events.length > 0;
+            return context;
+        }
```

Comment:
I'm wondering if there's a way to get rid of the `isTriggered` variable here. It doesn't feel like it should live on the `context` object, since it's only used in this class. It also feels slightly weird that you need to create the full context object before you can determine whether the event can be triggered.

It might be nice to not have this vague `if`, and instead have classes (or functions) encapsulating this logic at the time `TriggeredAbility` is constructed.

Reply:
Yeah so I actually realised that my implementation above _doesn't actually aggregate properly_; I could go into it but I'd be writing a full paragraph - I mostly noticed when I tried to convert Tywin (LoCR) to use aggregates and the whole system crumbled, so I switched this to a Draft and am trying to re-do it.

The next attempt is more doing what you recommend, where I'm making something like an `AggregateCondition` and `SingularCondition` for what events can trigger an ability. I'll send you a message if I have any issues, or when it's ready to review that update. :)

Totally agree with you on removing this `isTriggered` though - felt a little clunky!